packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,66 +1,225 @@
 import Objection from 'objection';
-import {SignedState, calculateChannelId, Objective} from '@statechannels/wallet-core';
+import {
+  SignedState,
+  Objective,
+  SignedStateWithHash,
+  SignedStateVarsWithHash,
+  Message,
+  hashState,
+  State,
+  calculateChannelId,
+} from '@statechannels/wallet-core';
+import _ from 'lodash';
+import {Either, left, right, isLeft} from 'fp-ts/lib/Either';
+import {Bytes32} from '@statechannels/client-api-schema';
 
 import {SignState} from '../protocols/actions';
 import {Channel, SyncState, RequiredColumns} from '../models/channel';
 import {SigningWallet} from '../models/signing-wallet';
-import {logger} from '../logger';
 import {addHash} from '../state-utils';
-import knex from '../db/connection';
 
 export const Store = {
-  transaction: async function(callback: () => Promise<void>): Promise<void> {
-    return knex.transaction(async () => {
-      await callback();
-    });
-  },
   signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
     const channel = await Channel.forId(action.channelId, tx);
-    const notifications = channel.signAndAdd(action);
-    await Channel.query(tx).update(channel);
+    const state: State = {...channel.channelConstants, ...action};
+
+    const validationResult = validateStateFreshness(state, channel);
+
+    if (isLeft(validationResult)) throw validationResult.left;
```

Comment:
Could re-use the `getOrThrow` helper by @andrewgordstewart here, and the renamed `isLeft` to `isError`.

Also I mentioned this [in a previous PR comment](https://github.com/statechannels/statechannels/pull/2368#discussion_r461212099), but we can clean up the responses using response value destructuring.

I still think we should do this... here is proof it works:

```ts
> let test = fp.either.right("abc")
undefined
> let { left: error, right : val } = example;
undefined
> if (error) throw Error('');
undefined
> val
'abc'
```

Reply:
Is that interpreted by node or ts-node? I would be surprised if that code compiles without type errors

Reply:
```
> import * as fp from 'fp-ts'
undefined
> let test = fp.either.right("abc")
undefined
> let { left: error, right : val } = test;
[eval].ts:7:7 - error TS2339: Property 'left' does not exist on type 'Either<never, string>'.

7 let { left: error, right : val } = test;
        ~~~~
[eval].ts:7:20 - error TS2339: Property 'right' does not exist on type 'Either<never, string>'.

7 let { left: error, right : val } = test;
```

It does not compile

Reply:
Sad :(

Reply:
It's not really that sad. We initially added `fp-ts` because it defined an `Either` type, and @kerzhner had good experience with it in the `simple-hub` package. We are free to define our own `Either` type that can have the property you'd like.

If we are not going to use `pipe`, `chain`, etc. from `fp-ts`, then maybe we just remove the dependency and define
```
type Either<E extends Error, T> = {error: E; value: undefined, ok: false} | {error: undefined, value: T, ok: true}

const test: Either<Error, 'test'> = {error: undefined, value: 'test', ok: true}
if (test.ok) {
  // test.value is of type `'test'`
}

if (test.error) {
  // test.error is of type `Error`
}

if (test.value) {
  // test.value is of type `'test'`
}
```

If we are going to use things like `pipe`, `chain`, etc. then maybe it helps to keep using `fp-ts`, and its `Either` type Or, even in that case, implement our own that extends `Either` from `fp`, but has the property that you want:
```
type Either<E, T> = {_tag: 'left'; left: E; right: undefined} | {_tag: 'right'; left: undefined; right: T}
```

Reply:
Do we want to throw an error in this case? It seems like we are mixing `Either`s and `throw`s. It looks like customers of the `signState` api ([one example](https://github.com/statechannels/statechannels/blob/ag/push-message/packages/server-wallet/src/wallet/index.ts#L116)), do not catch the error.

Reply:
Per https://statechannels.slack.com/archives/CPX2BU1FA/p1596063316027800, I think we should get rid of `Either` and use `throw`